core/storage/base_model/gae_models.py

```diff
@@ -295,6 +293,33 @@ def _reconstitute_from_snapshot_id(self, snapshot_id):
         reconstituted_model.last_updated = snapshot_model.last_updated
         return reconstituted_model
 
+    def _reconstitute_from_snapshot_id(self, snapshot_id):
+        """Makes this instance into a reconstitution of the given snapshot."""
+        snapshot_model = self.SNAPSHOT_CONTENT_CLASS.get(snapshot_id)
+        return self._reconstitute_from_snapshot_model(snapshot_model)
+
+    @classmethod
+    def _reconstitute_multi_from_models(cls, snapshot_id_model_list):
```

Comment:
I don't think I like snapshot_id_model_list -- it's a bit opaque. Can you pass in a list of dicts with keys that are human-understandable instead?